src/OrchardCore/OrchardCore.Logging.NLog/HostBuilderExtensions.cs

```diff
@@ -14,11 +15,16 @@ public static IHostBuilder UseNLogHost(this IHostBuilder builder)
         return builder
             .UseNLog()
             .ConfigureAppConfiguration((context, _) =>
+        {
+            var environment = context.HostingEnvironment;
+            var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);
+            var configDir = string.IsNullOrWhiteSpace(appData) ? Path.Combine(environment.ContentRootPath, ShellOptionConstants.DefaultAppDataPath) : appData;
+            if (LogManager.Configuration == null)
             {
-                var environment = context.HostingEnvironment;
-                var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);
-                var configDir = string.IsNullOrWhiteSpace(appData) ? $"{environment.ContentRootPath}/{ShellOptionConstants.DefaultAppDataPath}" : appData;
-                LogManager.Configuration.Variables["configDir"] = configDir;
-            });
+                LogManager.Configuration = new LoggingConfiguration();
```

Comment:
Maybe just perform null-check, instead of assigning dummy `LoggingConfiguration`:
```c#
var loggerConfiguration = LogManager.Configuration;
if (loggerConfiguration is not null)
    loggerConfiguration.Variables["configDir"] = configDir;
```

Reply:
No, if like your way, we cannot get log.
secondly,  LoggingConfiguration is not dummy .
after i add it:
1--no exception.
2--get the log

Reply:
When does this NRE happens?

Reply:
@Skrypt I'm only suggesting to remove `LogManager.Configuration = new LoggingConfiguration()`

Reply:
`Configuration` can only be null if the `NLog.config` is missing. I think we should follow what @snakefoot says.